app/views/spree/admin/products/new.html.haml

```diff
@@ -77,7 +77,7 @@
           = f.field_container :on_hand do
             = f.label :on_hand, t(".on_hand")
             %br/
-            = f.text_field :on_hand, class: 'fullwidth'
+            = f.text_field :on_hand, class: 'fullwidth', value: @on_hand
```

Comment:
I think I've already tried yes. 
And I just retried, and :
 - `on_hand` is filled with `0`, even if I fill the field with a value
 - Both `@product` and `@object` are the same object actually (I've printed, they both reference the same object `#<Spree::Product:0x00007fee3cfb7a48>`

So that's why I've found this way to fix the bug...